packages/cli/src/cli-commands/cli-version-commands.ts

```diff
@@ -81,7 +81,13 @@ export default {
       'dry-run': {
         describe: 'Displays the process command that would be performed without executing it.',
         group: 'Version Command Options:',
-        type: 'boolean',
+        coerce: (val: boolean | string) => {
```

Comment:
@Jason3S it seems that you were right and the argument next to dry-run is being assigned to the `bump`, I always thought that positional argument could only be provided as the first argument but that's not true and surprised about it. I passed `lerna version --no-changelog --dry-run premajor` and got the following (note that I did add back `type: 'boolean` on dry-run option)

![image](https://user-images.githubusercontent.com/643976/232847330-265df2f0-bf51-4298-9b6f-4a4cc33d3b77.png)

considering that it work as intended I wonder if we really need to throw an error on the dry-run then? Perhaps only the code change for the `conventional-prerelease` would be enough? What do you think?

Reply:
@ghiscoding,

I agree that the important change is to `conventional-prerelease`.

The reason for the `--dry-run` check is to catch behavior changes when `--dry-run` is removed.

`lerna command --option-with-optional-value --dry-run value`

In this case, removing `--dry-run` will change the behavior.

Reply:
> `lerna command --option-with-optional-value --dry-run value`
> 
> In this case, removing `--dry-run` will change the behavior.

sure but this is quite obvious that it is an incorrect usage of arguments, I don't think we need to worry about this though, it will never be able to catch every use case. 

So I think I'll just close this PR and just use the `conventional-prerelease` commit and extract it as a new PR, it's a good thing I made this PR into 2 separate commits